charger/em2go-home.go

```diff
@@ -0,0 +1,335 @@
+package charger
+
+// LICENSE
+
+// Copyright (c) 2019-2024 andig
+
+// This module is NOT covered by the MIT license. All rights reserved.
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import (
+	"encoding/binary"
+	"fmt"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/modbus"
+	"github.com/volkszaehler/mbmd/meters/rs485"
+)
+
+// https://www.em2go.de/download2/ModBus TCP Registers EM2GO Home Series.pdf
+
+// Em2Go charger implementation
+type Em2GoHome struct {
+	log  *util.Logger
+	conn *modbus.Connection
+}
+
+const (
+	em2GoHomeRegStatus			= 0   // Uint16 RO ENUM
+	em2GoHomeRegConnectorState	= 2   // Uint16 RO ENUM
+	em2GoHomeRegErrorCode		= 4   // Uint16 RO ENUM
+	em2GoHomeRegCurrent1		= 6   // Uint16 RO 0.1A
+	em2GoHomeRegCurrent2		= 8   // Uint16 RO 0.1A
+	em2GoHomeRegCurrent3		= 10  // Uint16 RO 0.1A
+	em2GoHomeRegPower			= 12  // Uint32 RO 1W
+	em2GoHomeRegPower1			= 16  // Uint16 RO 1W
+	em2GoHomeRegPower2			= 18  // Uint16 RO 1W
+	em2GoHomeRegPower3			= 20  // Uint16 RO 1W
+	em2GoHomeRegEnergy			= 28  // Uint16 RO 0.1KWh
+	em2GoHomeRegMaxCurrent		= 32  // Uint16 RO 0.1A
+	em2GoHomeRegMinCurrent		= 34  // Uint16 RO 0.1A
+	em2GoHomeRegCableMaxCurrent	= 36  // Uint16 RO 0.1A
+	em2GoHomeRegSerial			= 38  // Chr[16] RO UTF16
+	em2GoHomeRegChargedEnergy	= 72  // Uint16 RO 0.1kWh
+	em2GoHomeRegChargeDuration	= 78  // Uint32 RO 1s
+	em2GoHomeRegSafeCurrent		= 87  // Uint16 WR 0.1A
+	em2GoHomeRegCommTimeout		= 89  // Uint16 WR 1s
+	em2GoHomeRegCurrentLimit	= 91  // Uint16 WR 0.1A
+	em2GoHomeRegChargeMode		= 93  // Uint16 WR ENUM
+	em2GoHomeRegChargeCommand	= 95  // Uint16 WR ENUM
+	em2GoHomeRegVoltage1		= 109 // Uint16 RO 0.1V
+	em2GoHomeRegVoltage2		= 111 // Uint16 RO 0.1V
+	em2GoHomeRegVoltage3		= 113 // Uint16 RO 0.1V
```

Comment:
Das scheinen von der Adressierung her uint32 zu sein, die man ggf. in einem Rutsch lesen kann.